app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java

```diff
@@ -0,0 +1,250 @@
+            public int mOrientationDeg; //last rotation in degrees
+            private static final int _DATA_X = 0;
+            private static final int _DATA_Y = 1;
+            private static final int _DATA_Z = 2;
+            private int ORIENTATION_UNKNOWN = -1;
+
+            @Override
+            public void onSensorChanged(SensorEvent event)
+            {
+                float[] values = event.values;
+                int orientation = ORIENTATION_UNKNOWN;
+                float X = -values[_DATA_X];
+                float Y = -values[_DATA_Y];
+                float Z = -values[_DATA_Z];
+                float magnitude = X*X + Y*Y;
+                // Don't trust the angle if the magnitude is small compared to the y value
+                if (magnitude * 4 >= Z*Z) {
+                    float OneEightyOverPi = 57.29577957855f;
+                    float angle = (float)Math.atan2(-Y, X) * OneEightyOverPi;
+                    orientation = 90 - (int)Math.round(angle);
+                    // normalize to 0 - 359 range
+                    while (orientation >= 360) {
+                        orientation -= 360;
+                    }
+                    while (orientation < 0) {
+                        orientation += 360;
+                    }
+                }
+                //now we must figure out which orientation based on the degrees
+                if (orientation != mOrientationDeg)
+                {
+                    mOrientationDeg = orientation;
+
+                    View takePicButton = findViewById(R.id.take_photo_button);
+
+                    //figure out actual orientation
+                    if(orientation == -1){//basically flat
+                    }
+                    else if(orientation <= 45 || orientation > 315){//round to 0
+                        Log.d("Sensor", "P"); //Portrait
+                        orientationResult="P";
+                        takePicButton.setRotation(0); //rotate take picture button
+                    }
+                    else if(orientation > 45 && orientation <= 135){//round to 90
+                        Log.d("Sensor", "LR"); //LandscapeRight
+                        orientationResult="LR";
+                        takePicButton.setRotation(270);
+                    }
+                    else if(orientation > 135 && orientation <= 225){//round to 180
+                        Log.d("Sensor", "PU"); //PortraitUpside
+                        orientationResult="PU";
+                        takePicButton.setRotation(180);
+                    }
+                    else if(orientation > 225 && orientation <= 315){//round to 270
+                        Log.d("Sensor", "LL"); //LandscapeLeft
+                        orientationResult="LL";
+                        takePicButton.setRotation(90);
+                    }
+
+                }
+            }
+
+            @Override
+            public void onAccuracyChanged(Sensor sensor, int accuracy) {
```

Comment:
I didn't understand why is the method onAccuracyChanged  void?
Could be good to add a comment to explain that